### PR TITLE
Code review request for Library Project

### PR DESCRIPTION
**Pull request by fabricio-garcia**
This project includes:

- [x] All books stored in a simple array
- [x] An array hooked up to HTML with `render()`
- [x] It display books in a table, card, etc.
- [x] It has a "NEW BOOK" button that brings up a form allowing users to input the book details.
- [x] It has a button on each book’s display to remove the book from the library.
- [x] It has a button on each book’s display to change its read status.
- [x] Local Storage was used

captura_3

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,32 +1,111 @@
...
+  </script>
+  <script src="//cdnjs.cloudflare.com/ajax/libs/less.js/3.9.0/less.min.js"></script>
+  <script>
+    less.watch();
+  </script>
+</head>
+
+<body>
+  <header>
+    <img src="/src/assets/icons/research.png" alt="logo">
```
Comment (bolah2009) [inline index.html]: [Optional]

Let's add "." to the path or remove the first "/", currently, the images are not showing. See https://raw.githack.com/fabricio-garcia/library/6a210fa87583b62cf1603a2919de1d33b8e11a2d/index.html

```suggestion
    <img src="./src/assets/icons/research.png" alt="logo">
```

Let's do this for all images added to the app.

File: index.html [hunk 2]
```diff
@@ -1,32 +1,111 @@
...
+      <p>Share what books you read
+      </p>
     </div>
-  </body>
-</html>
+  </div>
+  <main>
+    <section class="notification" id="notification"></section>
+    <details>
+      <summary><img src="/src/assets/icons/bookadd.png" alt="add book"> New Book</summary>
```
Comment (bolah2009) [inline index.html]: [Optional]

See the previous comment

File: src/js/views.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    list.innerHTML = '';
+    myLibrary.forEach((book, id) => {
+      list.innerHTML += `
+      <tr key=${id} id="${book.isbn}">${book.view}</tr>
+      `;
+      document.body.addEventListener('click', (event) => {
+        // @ts-ignore
+        if (event.srcElement.id === `erase-${book.isbn}`) {
+          return addRemoveEvent(book);
+        }
```
Comment (bolah2009) [inline src/js/views.js]: Let's remove the book from the DOM when it's been erased. See below:

earsebook

File: src/js/templates.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    href="https://www.google.com/?q=%22${author}%22" 
+    target="_blank" rel="noopener noreferrer"
+    >
+    ${author}
+    <a/>
+  </td>
+  <td><span>ISBN:</span>${isbn}</td>
+  <td><span>Pages:</span>${pages}</td>
+  <td><span>Read?:</span>
+    <input type='checkbox' id="update-read-${isbn}"
```
Comment (bolah2009) [inline src/js/templates.js]: Let's make this a button instead and show the status if it has been `read` or not.

image

File: .eslintrc.json [hunk 5]
```diff
@@ -13,6 +13,13 @@
...
       "eol-last": "off",
-      "arrow-parens": "off"
+      "arrow-parens": "off",
+      "import/extensions": [
+        "error",
+        "ignorePackages",
+        {
+          "js": "always"
+        }
+]
```
[the remark this answers is not in the document]
Comment (Israel-Laguan) [reply .eslintrc.json]: The rules as they are implying that you must use just one file. By adding that extra rule you can use extra JS files to be more organized. 

Anyway, I erased the files and reorganized everything.

**SOLVED!**

File: index.html (hunk 1 above)
Comment (Israel-Laguan) [reply index.html]: **SOLVED!**

File: index.html (hunk 2 above)
Comment (Israel-Laguan) [reply index.html]: **SOLVED!**

File: src/js/views.js (hunk 3 above)
Comment (Israel-Laguan) [reply src/js/views.js]: I was using `display: none;` in order to "erase", now I use the method `.remove()` on the node.

**SOLVED!**

File: src/js/templates.js (hunk 4 above)
Comment (Israel-Laguan) [reply src/js/templates.js]: **Solved!**

**Comment by Israel-Laguan** [comment]
[The live version on last commit](https://raw.githack.com/fabricio-garcia/library/dee024abab2ebc4b84d7b10159bb424927deb6ee/index.html)

**Pull request merged**